### PR TITLE
mcux: scripts: pinctrl: update script to change RT 3 digital pinctrl model

### PR DESCRIPTION
**Pull request by lucien-nxp**
Change RT 3 digital platforms pinctrl model as below: 
index+offset+mux combined as a uint32_t

Regenerate pinctrl header files for RT500/600

### CODE REVIEW

File: mcux/scripts/pinctrl/gen_soc_headers.py [hunk 1]
```diff
@@ -117,8 +120,8 @@ def main():
 
     data_version = get_pack_version(temp_dir.name)
     print(f"Found data pack version {data_version}")
-    if round(data_version) != 14:
-        print("Warning: This tool is only verified for data pack version 14, "
+    if round(data_version) != 16:
```
Comment (danieldegrasse) [inline mcux/scripts/pinctrl/gen_soc_headers.py:123]: Thanks for updating this- can you make this change within a separate commit though?

**Review by danieldegrasse — APPROVED** [review]

**Review by dleach02 — APPROVED** [review]
